test/framework/interface.go

```diff
@@ -509,6 +509,9 @@ type Cluster interface {
 	DeployApp(fs ...AppDeploymentOption) error
 	Exec(namespace, podName, containerName string, cmd ...string) (string, string, error)
 	ExecWithRetries(namespace, podName, containerName string, cmd ...string) (string, string, error)
+	CreateNode(name string, label string) error
```

Comment:
Might be worth not adding this method here and only in the k8s version.

Reply:
And then do a multiple casts in the test? K8sCluster does not implement `WithRetries` and others, right?

Reply:
Yes and I think it's a mistake. These interfaces with 100s of method are not really the way go should be written and while it's annoying to fix I feel like we shouldn't make it worse. @jakubdyszkiewicz WDYT?

You can cast and store in a var and I believe this problem is absent on e2e_env because the cast is done elsewhere.

Reply:
I don't even use casting, I just re-use the variable returned by `NewK8sCluster`